merkle/src/merkle.rs

```diff
@@ -248,6 +252,158 @@ impl<E: Element> Clone for MmapStore<E> {
     }
 }
 
+/// File-mapping version of `MmapStore` with the added `new_with_path` method
+/// that allows to set its path (otherwise a temporary file is used which is
+/// cleaned up after we drop this structure).
+#[derive(Debug)]
+pub struct DiskMmapStore<E: Element> {
+    store: MmapMut,
+    len: usize,
+    _e: PhantomData<E>,
+    file: File,
+    // We need to save the `File` in case we're creating a `tempfile()`
+    // otherwise it will get cleaned after we return from `new()`.
+}
+
+impl<E: Element> ops::Deref for DiskMmapStore<E> {
+    type Target = [E];
+
+    fn deref(&self) -> &Self::Target {
+        unimplemented!()
+    }
+}
+
+impl<E: Element> Store<E> for DiskMmapStore<E> {
+    #[allow(unsafe_code)]
+    fn new(size: usize) -> Self {
+
+        // FIXME: CROSSING API BOUNDARIES HERE.
+        // This is used by `rust-fil-proofs` code when it allocates a `Store`
+        // to use it with `MerkleTree::from_data_with_store`, but the `Store`
+        // created there has an assigned size of *only* the leaves (the data,
+        // as it's seen there), not the *entire* tree, which is expected since
+        // the consumer shouldn't need to know how big the final `Store` is
+        // going to be. To *temporarily* accommodate this we *mangle* the
```

Comment:
I just want to make sure that by 'temporarily' you're referring to a stage of development, rather than an interval of program run-time. If that's the case, please ignore this comment. If not, we might want to discuss what's meant.

Reply:
Yes, this shouldn't land.